src/main/kotlin/com/sourcegraph/cody/history/state/ChatState.kt

```diff
@@ -23,15 +23,19 @@ class ChatState : BaseState() {
   @get:OptionTag(tag = "enhancedContext", nameAttribute = "")
   var enhancedContext: EnhancedContextState? by property()
 
+  private var updatedAtDate: LocalDateTime? = null
+
   fun title(): String? = messages.firstOrNull()?.text?.take(48)
 
   fun setUpdatedTimeAt(date: LocalDateTime) {
+    updatedAtDate = date
     updatedAt = date.format(DATE_FORMAT)
   }
 
   fun getUpdatedTimeAt(): LocalDateTime {
     if (updatedAt == null) return LocalDateTime.now()
-    return LocalDateTime.parse(updatedAt, DATE_FORMAT)
+    if (updatedAtDate == null) updatedAtDate = LocalDateTime.parse(updatedAt, DATE_FORMAT)
```

Comment:
```kotlin
  private var updatedAtDate: LocalDateTime = LocalDateTime.now()
  
  fun setUpdatedTimeAt(date: LocalDateTime) {
    updatedAt = date.format(DATE_FORMAT)
    updatedAtDate = date
  }

  fun getUpdatedTimeAt(): LocalDateTime {
      if (updatedAt == null) {
        setUpdatedTimeAt(LocalDateTime.now())
      }
      return updatedAtDate
  }
```

Reply:
Thanks! However, the chat dates are not properly initialized and everything lands in Today after deser. I had a similar problem, and as a result, it ended up with this code with `!!`.

![image](https://github.com/sourcegraph/jetbrains/assets/5013708/096f75e0-5238-4215-a776-31066f7f6b5b)

Reply:
Maybe let's just replace `return updatedAtDate!!` -> `return updatedAtDate ?: LocalDateTime.now()`?

Reply:
Yea, I realised afterwards it initially will be incorrectly initialized and will possibly not get re-initialized.
Do you know if that `BaseState` classes are initialised by constructor or using setters when loaded for the first time?